src/redux/books/books.js

```diff
@@ -0,0 +1,45 @@
+// to add action types
+const addBook = 'add_book';
+const removeBook = 'remove_book';
+
+// to initial the state array
+
+const initialState = [];
+
+// actions
+
+export const addBookAction = (id, title, author) => ({
+  type: addBook,
+  payload: {
+    id,
+    title,
+    author,
+  },
+});
+
+export const removeBookAction = (id) => ({
+  type: removeBook,
+  payload: id,
+});
+
+// the reducers
+const bookReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case 'addBook':
+      return [
+        ...state,
+        {
+          id: action.id,
+          title: action.title,
+          author: action.author,
+        },
```

Comment:
- [ ] Your reducer here will give an error here since there is no key called `action.id` but there is a key called `action.payload.id`. Just putting `action.payload` will automatically create a new book with the `id`, `title` and `author` so that will be even better.